ckanext/datadotworld/api.py

```diff
@@ -94,9 +94,10 @@ def notify(pkg_id):
 def _prepare_resource_url(res):
     """Convert list of resources to files_list for data.world.
     """
-    link = res['url']
-    name = res['name']
-
+    link = res['url'] or ''
```

Comment:
If what you want is `None` when `'url'` is a missing key, isn't `res.get('url')` what you want here?

Reply:
@rflprr as you can see this if is set for logging only. In CKAN those values should have values or be an empty string by default. The logging has been added to check if there is something that sets it to None in scheming or any other extension. As you can see from the PR name, it contains fixes as well as some debugging in case the issues are not fixed.

Sorry for missing your comment earlier.